Informatics I/010 Modularization and Git/Task 1 International Banking/Perytron Solution/bank_account.py

```diff
@@ -0,0 +1,34 @@
+from typing import Union
+from public.currency_converter import convert
+from public.exchange_rates import EXCHANGE_RATES
+
+class BankAccount:
+
+    def __init__(self, currency: str="CHF"):
+        self.__currency = self.__check_valid_currency(currency) # Check that currency exists
+        self.__balance = 0
+
+    def get_currency(self) -> str:
+        return self.__currency
+
+    def get_balance(self) -> Union[int, float]:
+        return self.__balance
+        
+    def deposit(self, amount: Union[int, float], currency="CHF") -> None:
+        # Add to the balance only if balance is positive afterwards and check that the amount and currency are actually provided in an allowed format.
+        self.__balance =  self.__check_zero(self.__balance + convert(amount, self.__check_valid_currency(currency), self.__currency))
```

Comment:
Good for making it water proof! (Hint: In case of additon can't become sup-zero)

Reply:
(could only happen if amount parameter is negative itself but this gets sorted out in convert, but if convert changes could be a possible bug, so good to be save!)